pkg/syncer/util.go

```diff
@@ -623,3 +625,89 @@ func getPVsInBoundAvailableOrReleasedForVc(ctx context.Context, metadataSyncer *
 
 	return k8svolumes, nil
 }
+
+// createStaticFileVolume attemts to create a static volume on each VC until it gets SUCCESS.
+// If while creating the volume, CNS returns CnsAlreadyRegisteredFault, it means that the volume does not need to be re-created.
+func createVolumeOnMultiVc(ctx context.Context, pv *v1.PersistentVolume,
+	metadataSyncer *metadataSyncInformer, volumeType string, metadataList []cnstypes.BaseCnsEntityMetadata, volumeHandle string) (string, volumes.Manager, error) {
+	log := logger.GetLogger(ctx)
+
+	vcconfigs, err := cnsvsphere.GetVirtualCenterConfigs(ctx, metadataSyncer.configInfo.Cfg)
+	if err != nil {
+		return "", nil, logger.LogNewErrorf(log, "failed to get VirtualCenterConfigs. err: %v", err)
+	}
+
+	for _, vcconfig := range vcconfigs {
+		log.Debugf("PVUpdated: Attempting to create volume on VC %s", vcconfig.Host)
+		volManager, err := getVolManagerForVcHost(ctx, vcconfig.Host, metadataSyncer)
+		if err != nil {
+			continue
+		}
+		err = createCnsVolume(ctx, pv, metadataSyncer, volManager, volumeType, vcconfig.Host, metadataList, volumeHandle)
+		if err == nil {
+			return vcconfig.Host, volManager, nil
+		}
+		log.Debugf("PVUpdated: Failed to create volume %s on VC %s", volumeHandle, vcconfig.Host)
```

Comment:
log the error here

Reply:
I am not logging error here because we are moving onto the next VC. Do you think it will be better to add Error instead?